packages/daemon/src/service/plugin.ts

```diff
@@ -128,23 +129,31 @@ export class PluginManager {
     return plugin;
   }
 
-  async install(pkg: PluginPackage, opts: InstallOptions): Promise<void> {
-    try {
-      await this.pluginRT.costa.install(pkg, opts);
-    } catch (err) {
-      // uninstall if occurring an error on installing.
-      await this.pluginRT.costa.uninstall(pkg.name);
-      throw err;
-    }
+  async install(pluginId: string, pkg: PluginPackage, opts: InstallOptions): Promise<void> {
+    return new Promise((resolve, reject) => {
+      pluginQueue.push((cb) => {
+        this.setStatusById(pluginId, PluginStatus.PENDING);
+        this.pluginRT.costa.install(pkg, opts).then(() => {
+          resolve();
+          cb();
+        }).catch((err) => {
+          // uninstall if occurring an error on installing.
+          this.pluginRT.costa.uninstall(pkg.name);
+          reject(err);
+          cb();
+        });
+      });
+    });
   }
 
-  async installAsync(pkg: PluginPackage, pyIndex?: string, force?: boolean): Promise<TraceResp<PluginResp>> {
+  async installNextTick(pkg: PluginPackage, pyIndex?: string, force?: boolean): Promise<TraceResp<PluginResp>> {
```

Comment:
Why change the function name?

Reply:
Because of the midway context scope, the name `nextTick` is a more specific  description and it's not going to affect the context scope.

Reply:
installAtNextTick?

Reply:
`installAtNextTick ` is better.